### PR TITLE
PERF: avoid copies if possible in fill_binop

### PR DESCRIPTION
**Pull request by jbrockmendel**
This has a small perf bump, but more importantly is necessary for the blockwise frame-with-frame PR that is coming up.

```
import pandas as pd
from pandas.core.ops import *

arr = np.arange(10**6)
df = pd.DataFrame({"A": arr})
ser = df["A"]

%timeit result = df.add(df, fill_value=4)
7.77 ms ± 20.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  <-- master
5.46 ms ± 36.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  <-- PR

%timeit result = ser.add(ser, fill_value=1)
6.79 ms ± 56.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  <-- master
4.65 ms ± 82.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  <-- PR
```

### CODE REVIEW

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -329,19 +329,27 @@ def fill_binop(left, right, fill_value):
...
-    Makes copies if fill_value is not None
+    Makes copies if fill_value is not None and NAs are present.
     """
-    # TODO: can we make a no-copy implementation?
     if fill_value is not None:
         left_mask = isna(left)
         right_mask = isna(right)
-        left = left.copy()
+
         right = right.copy()
```
Comment (WillAyd) [inline pandas/core/ops/__init__.py]: Should the right copy not be removed as well?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Not being super familiar here, is there a way to avoid copy without having to special case for the mask presence?

File: pandas/core/ops/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: good catch, will update

**Comment by jbrockmendel** [comment]
> Not being super familiar here, is there a way to avoid copy without having to special case for the mask presence?

Not sure I understand the question.  Can you give an example?

**Comment by gfyoung** [comment]
> Not sure I understand the question.

I believe @WillAyd is trying to determine if there is a faster way to figure out if we need to `copy` or not.  `.any` is still `O(n)`, so the average run-time is theoretically unchanged with your diff.

**Comment by jbrockmendel** [comment]
> determine if there is a faster way to figure out if we need to .copy or not

Well, we could pre-check for dtypes that can't hold NAs and short-circuit in those cases.

Other than that, we might be able to do 1 O(n) pass instead of 2-3 if we did it all in cython, but i dont think thats worthwhile at this point

**Comment by jreback** [comment]
this is fine. pls add asvs when you can.

**Pull request merged**